coral/src/app/components/FileInput.tsx

```diff
@@ -0,0 +1,149 @@
+          {labelText}
+          {props.required && <span className={"text-error-50"}>*</span>}
+        </Typography.Caption>
+      </Box>
+      <Grid
+        colGap={"l1"}
+        cols={"2"}
+        rows={"1"}
+        style={{
+          gridTemplateColumns: "max-content auto",
+        }}
+      >
+        <GridItem
+          colStart={"1"}
+          colEnd={"1"}
+          rowStart={"1"}
+          rowEnd={"1"}
+          width={"fit"}
+        >
+          <Box
+            aria-hidden={true}
+            display={"flex"}
+            alignItems={"center"}
+            backgroundColor={"white"}
+            className={`${classes.fakeButton}`}
+            onClick={handleWrapperClick}
+            data-testid="file-input-fake-button"
+          >
+            <Box
+              component={"div"}
+              display={"flex"}
+              alignItems={"center"}
+              colGap={"2"}
+              borderWidth={"1px"}
+              borderColor={"grey-30"}
+              borderRadius={"2px"}
+              paddingX={"l1"}
+              paddingY={"3"}
+              className={`${!valid && "border-error-50"}`}
+            >
+              <Icon icon={cloudUpload} />
+              <span>{buttonText}</span>
+            </Box>
+          </Box>
+        </GridItem>
+
+        <GridItem colStart={"2"} colEnd={"2"} rowStart={"1"} rowEnd={"1"}>
+          <Box
+            grow={"1"}
+            borderWidth={"1px"}
+            borderRadius={"2px"}
+            borderColor={"grey-20"}
+            backgroundColor={"grey-5"}
+            paddingX={"l1"}
+            paddingY={"3"}
+            aria-hidden={true}
+            data-testid="file-input-filename-info"
+          >
+            {inputRef.current?.files?.[0]?.name || noFileText}
+          </Box>
+        </GridItem>
+
+        <GridItem
+          className={classes.fileInputWrapper}
+          colStart={"1"}
+          colEnd={"1"}
+          rowStart={"1"}
+          rowEnd={"1"}
+        >
+          <label htmlFor={inputId}>
+            <span className={"visually-hidden"}>
+              {inputRef.current?.files?.[0]?.name
+                ? `Uploaded file, name: ${inputRef.current?.files?.[0]?.name}. Click to upload new file.`
+                : labelText}
+            </span>
+            <input
+              {...propsPassed}
+              id={inputId}
+              type={"file"}
+              ref={inputRef}
+              aria-required={props.required}
+              aria-invalid={`${!valid}`}
+              className={`${classes.fileInput}`}
+              {...(!valid && { "aria-describedby": errorMessageId })}
+            />
+          </label>
+        </GridItem>
+        <Box
+          component={"p"}
+          id={errorMessageId}
+          marginTop={"1"}
+          marginBottom={"3"}
+          className={"text-error-50 typography-caption-default"}
+        >
+          {valid ? <>&nbsp;</> : <>{helperText}</>}
```

Comment:
Maybe we should always render`{helperText}` as child of this `Box`? It should be `""` by default right?  Or is it semantically important to render a non breaking space? Like, the empty string might be read by some screen readers or something :o

Reply:
The non-breaking space is needed so the div has the same height as a div with text would have. Otherwise, there's a jump in the layout when an error appears. It does not have any effect on a screen reader!